### PR TITLE
Use AntialiasMode in CanvasPaintThread::new

### LINKED ISSUE #17661: Make CanvasPaintState::new accept an AntialiasMode enum argument instead of a boolean
Since it ends up using AntialiasMode internally, let's make it take the same argument type. We should also change the type of argument that CanvasPaintThread::new accepts as well.

Code: `components/canvas/canvas_paint_thread.rs`

No need for automated tests; if it compiles, that's good enough for a PR.

### PR DESCRIPTION
**Pull request by DarkDrek**
Change suggested in #17661

I only changed `canvas_paint_thread.rs` since `CanvasPaintThread::start` is used outside of the current crate and `azure::azure_hl::AntialiasMode` isn't available there.

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [X] These changes fix #17661 (github issue number if applicable).

- [ ] There are tests for these changes OR
- [X] These changes do not require tests because @jdm said so

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/17916)

### CODE REVIEW

File: components/canvas/canvas_paint_thread.rs [hunk 1]
```diff
@@ -127,6 +121,11 @@ impl<'a> CanvasPaintThread<'a> {
                  antialias: bool)
                  -> IpcSender<CanvasMsg> {
         let (sender, receiver) = ipc::channel::<CanvasMsg>().unwrap();
+        let antialias = if antialias {
```
Comment (emilio) [inline components/canvas/canvas_paint_thread.rs]: I wonder if we should pass this from the caller too.

**Review by emilio — APPROVED** [review]
Looks fine, thanks!

Next time, please ask in the issue to see if nobody is working on it, even if it's idle.

File: components/canvas/canvas_paint_thread.rs (hunk 1 above)
Comment (DarkDrek) [inline components/canvas/canvas_paint_thread.rs]: I also checked that but `azure::azure_hl::AntialiasMode` isn't available in the crate that uses it `components/constellation/constellation.rs` and I didn't want to change the .toml file for my first PR.

**Comment by emilio** [comment]
Sounds good! Thanks a lot for the PR :)

**Pull request merged**